components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMGroupResolver.java

```diff
@@ -528,11 +528,26 @@ public boolean listGroups(Condition condition, int limit, int offset, String dom
         }
         ExpressionCondition expressionCondition = (ExpressionCondition) condition;
         String attributeName = resolveGroupAttributeWithSCIMSchema(expressionCondition.getAttributeName(), tenantId);
-        String attributeValue = buildSearchAttributeValue(attributeName, expressionCondition.getOperation(),
-                expressionCondition.getAttributeValue(), SQL_FILTERING_DELIMITER);
+        String attributeValue;
+        if (StringUtils.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI, attributeName)) {
+            String nameWithDomain = expressionCondition.getAttributeValue();
+            if (!nameWithDomain.contains(CarbonConstants.DOMAIN_SEPARATOR)) {
+                nameWithDomain = domain + CarbonConstants.DOMAIN_SEPARATOR + nameWithDomain;
+            }
+            attributeValue = buildSearchAttributeValue(attributeName, expressionCondition.getOperation(),
+                    nameWithDomain, SQL_FILTERING_DELIMITER);
+        } else {
+            attributeValue = buildSearchAttributeValue(attributeName, expressionCondition.getOperation(),
+                    expressionCondition.getAttributeValue(), SQL_FILTERING_DELIMITER);
+        }
         GroupDAO groupDAO = new GroupDAO();
         try {
-            String[] groupNames = groupDAO.getGroupNameList(attributeName, attributeValue, tenantId, domain);
+            String[] groupNames;
+            if (StringUtils.equals(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME_URI, attributeName)) {
```

Comment:
This logic looks duplicated. Let's have a boolean variable to identify this new path of execution.

Reply:
addressed by https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/619/commits/7a68038b6bd8d48bf003136dda906e741f226cb6